include/sys/_pmap.h

```diff
@@ -0,0 +1,108 @@
+#ifndef _SYS__PMAP_H_
+#define _SYS__PMAP_H_
+
+#ifndef _MACHDEP
+#error "Do not use this header file outside kernel machine dependent code!"
+#endif
+
+#include <sys/mimiker.h>
+#include <sys/mutex.h>
+#include <sys/vm.h>
+#include <machine/pmap.h>
+
+struct pmap {
+  mtx_t mtx;                      /* protects all fields in this structure */
+  asid_t asid;                    /* address space identifier */
+  paddr_t pde;                    /* directory page table physical address */
+  vm_pagelist_t pte_pages;        /* pages we allocate in page table */
+  TAILQ_HEAD(, pv_entry) pv_list; /* all pages mapped by this physical map */
+
+  /* Machine-dependent part */
+  pmap_md_t md;
+};
+
+typedef struct pv_entry {
+  TAILQ_ENTRY(pv_entry) pmap_link; /* link on `pmap::pv_list` */
+  TAILQ_ENTRY(pv_entry) page_link; /* link on `vm_page::pv_list` */
+  pmap_t *pmap;                    /* page is mapped in this pmap */
+  vaddr_t va;                      /* under this address */
+} pv_entry_t;
+
+/*
+ * Machine dependent flags passed to `pmap_kenter` and `pmap_enter`.
+ */
+
+#define _PMAP_KERNEL (1 << 24)
```

Comment:
Flags should be in one place - it will be easier to read.

Reply:
I agree. Done.